include/veriblock/pop/blockchain/base_block_tree.hpp

```diff
@@ -519,6 +519,15 @@ struct BaseBlockTree {
     tips_.insert(index);
   }
 
+  //! dangerous! permanently drops the block from the tree
+  void eraseBlockUnchecked(index_t& block) {
```

Comment:
Maybe rename to deallocateBlockUnsafe?

Reply:
Why make a separate function for erasing block? You can merge `eraseBlock` and `eraseBlockUnchecked` into a single function.

Reply:
neither of eraseBlock functions is particularly safe. erase sounds unambiguous enough to me. drop might be also work. deallocate is a bit too low-level/implementation detail.

Reply:
one more point regarding "deallocate": what would you call this function in java?

Reply:
> Why make a separate function for erasing block? You can merge eraseBlock and eraseBlockUnchecked into a single function.
The point of multiple erase functions is contracts/invariant enforcement via asserts.

Reply:
> 

Well, eraseBlock should be eraseLeafBlock then. And eraseBlockUnchecked should be eraseAnyBlock.

Reply:
eraseBlock can erase the root block if the tree contains a single block, it's a bit of a mess atm. eraseBlockUnchecked was intended to sound unsafe as compared to eraseBlock.

Reply:
If the tree contains a single block then root block is also a leaf block.

Reply:
hmm, although we don't explicitly check that the block is not the root, we assert that block.isDeleted(), so in a tree that isn't malformed, we can't use the current implementation of eraseBlock() to erase the root block.

Reply:
anyway, this commit wasn't intended to make erase* stuff perfect: https://github.com/VeriBlock/alt-integration-cpp/pull/898#discussion_r759164702

Reply:
As to whether the root block is a leaf block, yes, formally it is. However, deleting the root block is not something the developer expects when manipulating leaf blocks. eraseRoot/eraseLeaf split is formally eraseRoot/eraseLeafThatIsNotRoot, but that's a bit too noisy.